shap/utils/_warnings.py

```diff
@@ -0,0 +1,6 @@
+class ExperimentalWarning(Warning):
```

Comment:
Decided to suggest custom warnings similar to existing `_exceptions.py`. Of course, it is not very consistent to add a new module in this commit, but this class can clearly be reused during the initial integration of new model and I feel blue placing it inside `_tree.py` :sweat_smile: .
Let me know if it is better to avoid it at this time.

Reply:
I think that's a good idea, let's go with it.